proposals/handler.go

```diff
@@ -272,25 +272,26 @@ func (h *Handler) handleProposal(ctx context.Context, expHash types.Hash32, peer
 	if p.Layer >= h.cfg.AllowEmptyActiveSet {
 		p.ActiveSet = nil
 	}
+
 	if !h.edVerifier.Verify(signing.PROPOSAL, p.SmesherID, p.SignedBytes(), p.Signature) {
 		badSigProposal.Inc()
 		return fmt.Errorf("failed to verify proposal signature")
 	}
-	p.ActiveSet = set
 	if !h.edVerifier.Verify(signing.BALLOT, p.Ballot.SmesherID, p.Ballot.SignedBytes(), p.Ballot.Signature) {
 		badSigBallot.Inc()
 		return fmt.Errorf("failed to verify ballot signature")
 	}
 
 	// set the proposal ID when received
+	// p.Initialize uses SignedBytes again
```

Comment:
It took me a while to understand why you moved active set assignment below. How about:
```suggestion
	// set the proposal ID when received
	// It mustn't contain the active set if layer >= AllowEmptyActiveSet
	// (p.Initialize uses SignedBytes again).
```